src/snowflake/cli/api/project/schemas/updatable_model.py

```diff
@@ -14,14 +15,29 @@ def __init__(self, *args, **kwargs):
         except ValidationError as e:
             raise SchemaValidationError(e)
 
-    def update_from_dict(
-        self, update_values: Dict[str, Any]
-    ):  # this method works wrong for optional fields set to None
-        for field, value in update_values.items():  # do we even need this?
-            if getattr(self, field, None):
-                setattr(self, field, value)
+    def update_from_dict(self, update_values: Dict[str, Any]) -> object:
+        for field, value in update_values.items():
+            if field in self.model_fields.keys():
+                if (
+                    self._is_field_an_updatable_model(field)
+                    and field in self.model_fields_set
+                ):
+                    getattr(self, field).update_from_dict(value)
+                else:
+                    setattr(self, field, value)
         return self
 
+    def _is_field_an_updatable_model(self, name: str) -> bool:
+        typing_types = (Literal, list)
+        field_type = self.model_fields.get(name, None)
+        if field_type:
+            if (args := get_args(field_type.annotation)) and not any(
+                [get_origin(arg) in typing_types for arg in args]
+            ):
+                return any([issubclass(arg, UpdatableModel) for arg in args])
+            return issubclass(type(field_type.annotation), UpdatableModel)
+        return False
```

Comment:
Could you explain what this function do?

Reply:
I'm trying to decide, if i should just update value, or recursively call `update_from_dict` method on a field that is a Pydantic schema itself.

The problem i encountered, was that some fields are optional, therefore the 'self.model_fields.get(name)` does return more complex type i have to parse.

Reply:
Let's document this in code

Reply:
Done